File: cpp/01/main.cpp
#include <fstream>
#include <iostream>
#include <string>

using namespace std;

struct Node {
public:
  int data;
  Node *next;

  Node(int data) {
    this->data = data;
    next = NULL;
  }

  void add(int data);
};

void Node::add(int data) {
  Node *n = new Node(data);
  if (next == NULL) {
	next = n;
	return;
  }

  Node *current = next;
  while (current->next != NULL) {
    current = current->next;
  }
  current->next = n;
}

int main() {

  ifstream myfile("./input.txt");
  string line;

  Node *head = NULL;

  if (myfile.is_open()) {
    while (getline(myfile, line)) {
      if (line != "") {
        int data = stoi(line);
		  if (head == NULL)
			  head = new Node(data);
		  else
        	head->add(data);
      }
    }

    myfile.close();
  } else {
    cout << "Unable to open file\n";
  }

  Node *current = head;
  int prev = INT_MAX; 
  int count = 0;
  while (current != NULL) {
	  if (current->data > prev) {
		  count++;
	  }
	prev = current->data;
    current = current->next;
  }

  cout << "Total Count: " << count << "\n";

  return 0;
}

File: go/01/main.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"strings"
	"strconv"
)

func main() {
	body, err := ioutil.ReadFile("./input.txt")
	if err != nil {
		log.Fatalf("unable to read file: %v", err)
	}

	data := strings.Split(string(body), "\n")
	prev, err := strconv.Atoi(data[0])

	if err != nil {
		log.Fatalf("could not parse %v to int", data[0])
	}

	count := 0

	for i := 0; i < len(data) - 3; i++{
		current, err := strconv.Atoi(data[i + 3])

		if err != nil {
			continue
		}

		if (current > prev) {
			count++
		}
		prev, err = strconv.Atoi(data[i + 1])
	}

	fmt.Println(count)
}
